### PR TITLE
Add batched rotmg

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims at adding a batched interface for [rotmg](https://www.netlib.org/lapack/explore-html/d3/dd5/group__rotmg_gaebf62f1c90f0829a0a762a7f8918213f.html#gaebf62f1c90f0829a0a762a7f8918213f). 

- [x] Refactor `rotmg_impl` under blas. Fixed the behavior for `d1 < 0`.
- [x] Add batched interface `Rotmg`. `SerialRotmg` and `TeamRotmg` are not added because this kernel works only on scalars without any parallelization
- [x] Tests introduced under batched. Some corner case (`d1 < 0`) is not covered

### CODE REVIEW

File: blas/impl/KokkosBlas1_rotmg_impl.hpp
Comment (lucbv) [inline blas/impl/KokkosBlas1_rotmg_impl.hpp:1]: Can you explain to me why you are removing all the views here?
Typically we use raw pointers so we can mark them as `KOKKOS_RESTRICT` however rotmg computes only four values and cannot be vectorized...
I would recommend keeping the view for this one so we can benefit from ref counting and debug bounds checking.

File: batched/dense/impl/KokkosBatched_Rotmg_Impl.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    !KokkosKernels::ArithTraits<p_value_type>::is_complex,
+                "KokkosBatched::rotmg: Complex types are not supported for Rotmg.");
+#ifndef NDEBUG
+  // param should include flag, h11, h21, h12, h22
+  if (param.extent_int(0) != 5) {
+    Kokkos::printf("KokkosBatched::rotmg: param must have length 5: param length = %d\n", param.extent_int(0));
+    return 1;
+  }
+#endif
+  KokkosBlas::Impl::rotmg_impl(d1.data(), d2.data(), x1.data(), y1.data(), param.data(), param.stride(0));
```
Comment (lucbv) [inline batched/dense/impl/KokkosBatched_Rotmg_Impl.hpp]: Keep the views here

**Review by lucbv — CHANGES_REQUESTED** [review]
I am pretty much okay with this, I just do not want to change from View to raw pointers in the Blas impl file since we cannot vectorize in this function.

File: blas/impl/KokkosBlas1_rotmg_impl.hpp
Comment (yasahi-hpc) [reply blas/impl/KokkosBlas1_rotmg_impl.hpp:1]: OK
I made them pointers in order to align with `rotg_impl` implementation where the values are represented by pointers. 
I do not have a strong preference, so I can make them views as you want

**Review by lucbv — APPROVED** [review]
